### ISSUE TITLE
ocaml linking error

### ISSUE THREAD

**Issue by icristescu** [Z3Prover/z3#724]
I have installed Z3 4.4.1 following the instructions in the readme file. I used 
```
python scripts/mk_make.py --ml
```
to have the ocaml bindings. 

In the readme file in directory examples/ml it states that in order to obtain the bytecode of ml_example.ml I have to use the following command: ```ocamlfind ocamlc -o ml_example.byte -package Z3 -linkpkg ml_example.ml```

I have the following error: 
```
Error on dynamically loaded library: /Users/ioana/.opam/system/lib/stublibs/dllz3ml.so: dlopen(/Users/ioana/.opam/system/lib/stublibs/dllz3ml.so, 138): Library not loaded: libz3.dylib
  Referenced from: /Users/ioana/.opam/system/lib/stublibs/dllz3ml.so
  Reason: image not found
```
(Building the native code version works fine)
I also tried to run the command (I'm on a mac); ```DYLD_LIBRARY_PATH=/Users/ioana/.opam/system/lib/Z3/ ocamlfind ocamlc -o ml_example.byte -package Z3 -linkpkg ml_example.ml```
 
and get the following:
```
Error while linking /Users/ioana/.opam/system/lib/Z3/z3ml.cma(Z3native):
The external function `n_is_null' is not available
```

**Comment by wintersteiger** [comment]
Thanks or reporting this issue. The bytecode version of the OCaml API is particularly fragile. Whether it works or not is highly dependent on your system and the versions of OCaml and findlib, etc. Could you tell us which version of each of those tools you are using and the precise version of Z3 you're using (e.g., release, nightly binary, or source from master), as well as the OSX version and your default compiler for C/C++?

**Comment by martin-neuhaeusser** [comment]
I just checked with the latest master (3b70dd6) on Debian 8, amd64 and the bytecode compile succeeds and produces a working binary.

Could it be that you are using an outdated version of the OCaml bindings? I removed the function `n_is_null` completely during an update of the OCaml API. It has been replaced by type safe versions which are called `n_is_null_symbol`, `n_is_null_ast`, etc. 
The symbol `n_is_null` does no longer exist.
